src/components/safe-apps/AppFrame/useAppCommunicator.ts

```diff
@@ -0,0 +1,150 @@
+import { MutableRefObject, useEffect, useMemo, useState } from 'react'
+import { getAddress } from 'ethers/lib/utils'
+import { getBalances, getTransactionDetails, SafeAppData } from '@gnosis.pm/safe-react-gateway-sdk'
+import {
+  GetBalanceParams,
+  GetTxBySafeTxHashParams,
+  Methods,
+  RPCPayload,
+  SendTransactionsParams,
+  SignMessageParams,
+} from '@gnosis.pm/safe-apps-sdk'
+import AppCommunicator from '@/services/safe-apps/AppCommunicator'
+import { Errors, logError } from '@/services/exceptions'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import useIsGranted from '@/hooks/useIsGranted'
+import { useCurrentChain } from '@/hooks/useChains'
+import useSafeAddress from '@/hooks/useSafeAddress'
+import { createSafeAppsWeb3Provider } from '@/hooks/wallets/web3'
+
+type JsonRpcResponse = {
```

Comment:
I think this type should be as well in some package (ethers?) but cannot find it